### PR TITLE
improved automatic search of subtitles for video without metadata

### PR DESCRIPTION
**Pull request by emavgl**
I've touched the lines below 219
`if item['title'] == "":`
to improve automatic search of subtitles whenever there is a video without correct metadata.
Using a regex the code parses the name of the file and gets: name of the show, episode and season.
Regex matches the most commons subtitles string forms like:
- Gotham S02E03
- Gotham 2x04  
- ecc

### CODE REVIEW

File: service.py [hunk 1]
```diff
@@ -201,14 +215,29 @@ def get_params():
   item['tvshow']=checkexp(item['tvshow'])
   for lang in urllib.unquote(params['languages']).decode('utf-8').split(","):
     item['3let_language'].append(xbmc.convertLanguage(lang,xbmc.ISO_639_2))
-  
+
   if item['title'] == "":
     item['title']  = normalizeString(xbmc.getInfoLabel("VideoPlayer.Title"))      # no original title, get just Title
-    
+    toParse = item['title'].lower().replace('.', " ");
+    res=re.findall('(.*?)s?0?(\d{1,3})x?e?0?(\d{1,3})', urllib.unquote(toParse), re.IGNORECASE)
```
Comment (luivit) [inline service.py]: Use **parseSearchString** even here

File: service.py [hunk 2]
```diff
@@ -201,14 +215,29 @@ def get_params():
...
+        item['tvshow']=res[0][0]
+        lres=len(item['tvshow'])
+        if item['tvshow'][lres-1:lres]!=" ":
+            item['tvshow']=item['tvshow']+" "
+        item['season']=res[0][1]
+        item['episode']=res[0][2]
+        item['3let_language']=[]
+        for lang in urllib.unquote(params['languages']).decode('utf-8').split(","):
+            item['3let_language'].append(xbmc.convertLanguage(lang,xbmc.ISO_639_2))
+        Search(item)
```
Comment (luivit) [inline service.py]: remove the 233 line (otherwise it is called 2 times)
thx

**Comment by emavgl** [comment]
Should be fixed in 902288852cb186a15efd80befa9b1e060cb27748
Give it a look.

**Comment by luivit** [comment]
yes, now the code it's ok but regex doesn't work with tvshow that have numbers in title.
(es. the 100, the 4400, 100 Questions, 11.22.63, 12 Monkeys, 1600 Penn, ecc)
you should edit into:
`(.*\d*?)s?0?(\d{1,3})x?e?0?(\d{1,3})`

**Comment by emavgl** [comment]
I think that your regex doesn't work too: I tried it on [rubular](http://rubular.com/) and the "Match groups" list isn't what we expected. You can try yourself using the text: _The 100 s01e03_  
I've added a space on it between `(.*\d*?)` and the rest of the regex. It should work now.

Fixed here e3b7e83a842af8e5a0ec4758f5b00e1729ff202b

**Comment by luivit** [comment]
yes, i forgot the space while i was writing regex here :)

**Pull request merged**